pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh

```diff
@@ -0,0 +1,76 @@
+npmSetupHook() {
+    echo "Executing npmSetupHook"
+
+    runHook preConfigure
+
+    echo "Configuring npm"
+
+    export HOME=$TMPDIR
+    export npm_config_nodedir="@nodeSrc@"
+
+    npm config set cache "$npmDeps"
+    npm config set offline true
+    npm config set progress false
+
+    local cacheLockfile="$npmDeps/package-lock.json"
+    local srcLockfile="$PWD/package-lock.json"
```

Comment:
This should be fine too.
```suggestion
    local srcLockfile="./package-lock.json"
```

I also think it would potentially be benefitial to make the location overridable, something like this:
```bash
    if [ -z "${nmpLockfile-}" ]; then
        npmLockfile="./package-lock.json"
    fi
```

Reply:
> This should be fine too.

I disagree -- using `$PWD` makes it more clear in the logs what copy we're comparing against.

Reply:
Ah right, that's a good point. We could also use `$(realpath "./package-lock.json")` but i suppose that's a bit wordier for the same result.

I won't mark this as resolved for now since the other comment does still apply.